draco/analysis/sidereal.py

```diff
@@ -703,8 +703,11 @@ def process(self, sdata):
             self.stack.weight[:] = 0.0
 
             self.count = 0
-            self.Ni_s = np.zeros(
-                (sdata.weight.local_shape[0], sdata.weight.local_shape[2])
+            self.Ni_s = mpiarray.zeros(
```

Comment:
Does this need to be an mpiarray? I don't think it goes through any redistrbutions

Reply:
I agree. On reflection I don't think this needs to be an MPIArray. That said, I'm going to leave it as one because it currently works and I don't want to break it when I change it.